### DISCUSSION TITLE
Why is ${DOCUMENTS} not recognized in profile?

### DISCUSSION THREAD

**Discussion by realnc** [netblue30/firejail#5166] [Q&A] [answered]
I've been using firejail to run discord for a while now. Recently I discovered that this didn't do anything to prevent access to my files that I keep in `/mnt/Data` on a different storage device. So now I launch discord in firejail 0.9.68 with this `~/.config/firejail/discord.local`:

```text
ignore private-bin
ignore private-opt
ignore private-etc
noblacklist ${DOCUMENTS}
blacklist /mnt/Data/*
```

(The `ignore` lines are required because without them, clicking links in discord does nothing. They don't open in the browser.)

I have to manually use `blacklist /mnt/Data/*` because apparently firejail will happily give discord access to all my files. But the `noblacklist ${DOCUMENTS}` line doesn't work. I have no access to documents. The output of `xdg-user-dir` is:

```
xdg-user-dir DOCUMENTS
/mnt/Data/Documents
```

If I use:

```
noblacklist /mnt/Data/Documents
```

Then it works fine. Does someone know why?

**Comment by rusty-snake** [comment, marked as answer]
> (The ignore lines are required because without them, clicking links in discord does nothing. They don't open in the browser.)

OT: There are safer alternatives, follow links from https://github.com/netblue30/firejail/issues/3919#issuecomment-768078226.

> Does someone know why?

xdg-user-dirs macros are implemented in a way that assumes they are in `$HOME`, see #3735.

**Reply by realnc** [reply]
> xdg-user-dirs macros are implemented in a way that assumes they are in `$HOME`

I have symlinks in $HOME. `~/Documents` points to `/mnt/Data/Documents`. Is this supposed to work? It doesn't seem to. It works for `~/Downloads` though, which is weird.

**Reply by rusty-snake** [reply]
> Is this supposed to work?

You mean that `noblacklist ${HOME}/Documents` `noblacklist`s `/mnt/Data/Documents`? No, `noblacklist` is implemented as path-comparison when applying/parsing `blacklist` and does not look at the filesystem.

**Reply by realnc** [reply]
> You mean that `noblacklist ${HOME}/Documents` `noblacklist`s `/mnt/Data/Documents`?

Nope. It's `noblacklist /mnt/Data/Downloads` that allows access to `~/Downloads`. But `noblacklist /mnt/Data/Documents` still won't allow access to `~/Documents`.

**Reply by rusty-snake** [reply]
`blacklist ${DOCUMENTS}` in disable-xdg.inc?

**Reply by realnc** [reply]
> `blacklist ${DOCUMENTS}` in disable-xdg.inc?

Shouldn't `noblacklist ${DOCUMENTS}` undo that? Also, there's `ignore include disable-xdg.inc` in `/etc/firejail/discord-common.profile`.

**Reply by realnc** [reply]
Never mind. I needed `whitelist ${HOME}/Documents` and now both access through the `~/Documents` symlink as well as `/mnt/Data/Documents` works fine.
